### PR TITLE
Fixed GoSec failing tests

### PR DESCRIPTION
**Pull request by ManjunathMS35**
#873

### CODE REVIEW

File: src/main/java/com/sap/oss/phosphor/fosstars/data/AbstractStaticScanToolsDataProvider.java [hunk 1]
```diff
@@ -183,13 +183,14 @@ public static List<Path> findIniConfigsIn(LocalRepository repository) throws IOE
    * Browse through the GitHub actions and find matches with the given predicates.
    *
    * @param githubAction GitHub Actions.
-   * @param matchers predicates to match through the given action.
-   * @return True if one of the predicate finds matches. False otherwise.
+   * @param matchers     predicates to match through the given action.
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/oss/phosphor/fosstars/data/AbstractStaticScanToolsDataProvider.java:186]:
```suggestion
   * @param matchers predicates to match through the given action.
```

File: src/main/java/com/sap/oss/phosphor/fosstars/data/AbstractStaticScanToolsDataProvider.java [hunk 2]
```diff
@@ -199,38 +200,32 @@ private static boolean scanGitHubAction(Map<?, ?> githubAction,
...
-  private static boolean scanSteps(Iterable<?> steps, Map<String, Predicate<String>> matchers) {
-    return IteratorUtils.toList(steps.iterator()).stream()
+        .filter(ArrayList.class::isInstance)
+        .map(ArrayList.class::cast)
+        .flatMap(step -> step.stream())
         .filter(Map.class::isInstance)
         .map(Map.class::cast)
-        .anyMatch(step -> MATCH_MAP_PREDICATE.test(step, matchers));
+        .filter(step -> MATCH_MAP_PREDICATE.test((Map<?, ?>) step, matchers))
+        .findAny();
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/oss/phosphor/fosstars/data/AbstractStaticScanToolsDataProvider.java:228]: @ManjunathMS35 I think, I have fixed it. Please review it and check if we can optimize this.

Comment (ManjunathMS35) [reply src/main/java/com/sap/oss/phosphor/fosstars/data/AbstractStaticScanToolsDataProvider.java]: @sourabhsparkala The solution looks good to me. If there is nothing to change in this PR could please approve it?

**Review by sourabhsparkala — APPROVED** [review]
👍

**Pull request merged**